Improve Documentation as per Issue #248

kilnhast opened the pull request:
This PR addresses the documentation improvement requested in [#248](https://github.com/trailofbits/test-fuzz/issues/248).

Added two important clarifications to the `cargo test-fuzz` command documentation section:

1. Added information that output written to stderr by the target is shown by the `--display` option
2. Added information that the `--display` and `--replay` options can be passed together

These additions provide clearer guidance for users about how to view error output and combine these command options effectively.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/trailofbits/test-fuzz?pullRequest=537) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/trailofbits/test-fuzz?pullRequest=537) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/trailofbits/test-fuzz?pullRequest=537) it.</sub>

smoelius commented:
It looks like you are getting some Clippy failures that are not the fault of your PR. I will try to fix those within the next few days.

smoelius commented:
Rebasing onto the latest master should make the Clippy warnings go away.

kilnhast commented:
@smoelius  
I have added the following changes as per your Suggestions 

- Added information that output written to stderr is shown by --display
- Added information that --display and --replay can be passed together
- Placed additions in the "cargo test-fuzz command" section (not in "Overview")
- Formatted as paragraphs rather than enumerated elements
- Mentioned dbg! in the stderr paragraph
- also, I have put the code back in referred position:

     ```
[deriving them]: https://serde.rs/derive.html
     [is not enabled]: https://github.com/rust-lang/rust/issues/45599#issuecomment-460488107
     [patch]: https://doc.rust-lang.org/edition-guide/rust-2018/cargo-and-crates-io/replacing-dependencies-with-patch.html
     [won't allow you to]: https://doc.rust-lang.org/book/ch19-03-advanced-traits.html#using-the-newtype-pattern-to-implement-external-traits-on-external-types
```

kilnhast commented on README.md:
```diff
@@ -521,75 +533,4 @@ These options are incompatible in the following sense. If a fuzz target's argume
-[deriving them]: https://serde.rs/derive.html
-[is not enabled]: https://github.com/rust-lang/rust/issues/45599#issuecomment-460488107
-[patch]: https://doc.rust-lang.org/edition-guide/rust-2018/cargo-and-crates-io/replacing-dependencies-with-patch.html
-[won't allow you to]: https://doc.rust-lang.org/book/ch19-03-advanced-traits.html#using-the-newtype-pattern-to-implement-external-traits-on-external-types
```
Yes I have put them back in with the Latest commit I have made

smoelius reviewed (commented):
If you could please squash your commits into one after making these changes, I think this should be good to go.

I really appreciate you working on this.

smoelius commented on README.md:
````diff
@@ -276,6 +276,7 @@ The `cargo test-fuzz` command is used to interact with fuzz targets, and to mani
    ```
 
 4. Replay crashes found for target `foo`
+
````
```suggestion
```

kilnhast commented:
@smoelius I have squashed all of my previous commits into one single commit with the suggested changes too. Can you review and if it is done correctly, can you merge the PR ?

smoelius commented:
@kilnhast Sorry, but it looks like some suggestions still haven't been applied. (I know I am guilty of making [one suggestion](https://github.com/trailofbits/test-fuzz/pull/537#discussion_r2023827278) late.)

Could you please take another look?

kilnhast commented:
@smoelius , I sincerely apologize for the error. It seems I mistakenly reverted the changes and then squashed all the commits into one. I have now applied the suggestions accordingly and made the commit. Could you review if everything is correct now? If so, could you merge the PR?

smoelius commented:
I think these changes may have gotten lost? (I liked having the code example.)
- https://github.com/trailofbits/test-fuzz/pull/537#discussion_r2023665302
- https://github.com/trailofbits/test-fuzz/pull/537#discussion_r2023665538

If you just push one commit on top of what you have, that should be fine.

kilnhast commented:
@smoelius ,  I’ve added the code example to the documentation as you suggested. Thank you for sharing your preferences and for your helpful feedback on my errors!

smoelius commented:
Thanks a lot!
